app/components/filters/select-multiple-filter.ts

```diff
@@ -1,29 +1,58 @@
 import { action, get } from '@ember/object';
 import { tracked } from '@glimmer/tracking';
-import {
-  deserializeArray,
-  serializeArray,
-} from 'frontend-burgernabije-besluitendatabank/utils/query-params';
+import { deserializeArray } from 'frontend-burgernabije-besluitendatabank/utils/query-params';
 import FilterComponent, { type FilterArgs } from './filter';
 
 type Option = Record<string, string>;
 
+type GroupedOptions = {
+  groupName: string;
+  options: Option[];
+};
+
 interface Signature {
   Args: {
+    route?: string;
     options: Promise<Option[]>;
   } & FilterArgs;
 }
 
 export default class SelectMultipleFilterComponent extends FilterComponent<Signature> {
   @tracked selected?: Option[];
 
-  /** Action to parse the queryParameter value(s) & select them in the select input */
   @action
   async inserted() {
-    // The queryParam is an array of searchField values, joined by seperator
-    // On the page load, we can use this to load in values
-    const queryParam = this.getQueryParam(this.args.queryParam);
-    if (queryParam) {
+    if (this.args.queryParam?.includes(',')) {
+      const queryParams = this.getQueryParam(this.args.queryParam) as string;
+      const needles = queryParams.split(/[+,]/);
+      const searchField = this.args.searchField;
+
+      const haystack = (await this.args.options) as unknown as GroupedOptions[];
+      const results: Option[] = [];
+
+      const flattenedHaystack: Option[] = [];
+      if (haystack[0]?.['groupName']) {
+        haystack.forEach((group) => {
+          group['options'].forEach((option: Option) => {
+            flattenedHaystack.push(option);
+          });
+        });
+      }
+
+      for (let i = 0; i < needles.length; i++) {
+        const needle = needles[i];
+        const found = flattenedHaystack.find(
+          (value) => get(value, searchField) === needle
+        );
+        if (found) {
+          results.push(found);
+        }
+      }
+
+      this.selected = results;
+    } else {
+      const queryParam = this.getQueryParam(this.args.queryParam) as string;
```

Comment:
**Blocker**: You do not handle the case, when `this.args.queryParam` is undefined.

Reply:
It doesn't crash if it is undefined, am I supposed to still add a handler for it then?

Reply:
Sorry, it is not when `this.args.queryParam` but when `this.getQueryParam(this.args.queryParam)`return `undefined`. The previous version of the code was handling this case. you can simply add a guard like `if(!queryParam) return [];